src/common/src/system_param/local_manager.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ops::Deref;
+use std::sync::Arc;
+
+use arc_swap::ArcSwap;
+use risingwave_pb::meta::SystemParams;
+use tokio::sync::broadcast::{channel, Sender};
+
+use super::reader::SystemParamsReader;
+use crate::util::channel_util::broadcast::IgnoreLaggedReceiver;
+
+pub type SystemParamsReaderRef = Arc<ArcSwap<SystemParamsReader>>;
+
+/// The system parameter manager on worker nodes. It provides two methods for other components to
+/// read the latest system parameters:
+/// - `get_params` returns a reference to the latest parameters that is atomically updated.
+/// - `subscribe_params` returns a channel on which calling `recv` will get the latest parameters.
+///   Compared with `get_params`, the caller can be explicitly notifed of parameter change.
+pub struct LocalSystemParamManager {
+    /// The latest parameters.
+    params: SystemParamsReaderRef,
+
+    /// Sender of the latest parameters.
+    tx: Sender<SystemParamsReaderRef>,
+}
+
+impl LocalSystemParamManager {
+    pub fn new(params: SystemParamsReader) -> Self {
+        // Only care about latest params.
+        let (tx, _) = channel(1);
```

Comment:
If we only want the latest value from the channel, will the `watch` channel be a better choice? https://docs.rs/tokio/latest/tokio/sync/watch/index.html

Reply:
That's exactly what I need